### DISCUSSION TITLE
Real world usage of jiff::civil::DateTime

### DISCUSSION THREAD

**Discussion by tisonkun** [BurntSushi/jiff#135] [Q&A] [answered]
Despite of legacy code, I'm a bit confused when jiff::civil::DateTime is preferred than Timestamp or Zoned.

Since a valid timestamp(datatime) should be always with the time zone, or else it's hard to determinate when it accurately represents.

**Comment by BurntSushi** [comment, marked as answer]
It is very difficult for me to understand your question. Do the [civil module docs](https://docs.rs/jiff/latest/jiff/civil/index.html) help? If not, what specifically is confusing to you? 

> Since a valid timestamp(datatime) should be always with the time zone

A Jiff `Timestamp` is just a 96-bit integer of nanoseconds since the Unix epoch. It has no time zone. 

A time zone is something like `America/New_York`. It is a separate piece of information. A time zone is not required to represent an instant in time. 

Given a civil time, however, a time zone is required in order to turn that civil time into an instant in time. Otherwise, the civil time is ambiguous. How do you know how to interpret it?

Maybe these docs would also help: https://docs.rs/jiff/latest/jiff/_documentation/platform/index.html#vocabulary

**Reply by tisonkun** [reply]
Thank you! I recently build a database and try to understand what datetime type is reasonable to expose to users for manipulations.

It looks to me that civil time comes from the day when timezone hadn't been integrated into timestamp generally. And now db users would better use timstamp or zoned instead.

The module doc and vocabulary give the information helpful for my scenario. I should first try to search the docs first 🤣
